anndata/tests/test_concatenate_disk.py

```diff
@@ -0,0 +1,238 @@
+from typing import Mapping
+
+import numpy as np
+import pandas as pd
+import pytest
+from scipy import sparse
+
+from anndata._io.merge import concat_on_disk
+from anndata import AnnData, concat
+from anndata.tests.helpers import (
+    assert_equal,
+    gen_adata,
+)
+
+
+from anndata.utils import asarray
+
+
+from anndata import read_h5ad, read_zarr
+
+
+GEN_ADATA_OOC_CONCAT_ARGS = dict(
+    obsm_types=(
+        sparse.csr_matrix,
+        np.ndarray,
+        pd.DataFrame,
+    ),
+    varm_types=(sparse.csr_matrix, np.ndarray, pd.DataFrame),
+    layers_types=(sparse.spmatrix, np.ndarray, pd.DataFrame),
+)
+
+
+@pytest.fixture(params=[0, 1])
+def axis(request):
+    return request.param
+
+
+@pytest.fixture(
+    params=["array", "sparse"],
+)
+def array_type(request):
+    return request.param
+
+
+@pytest.fixture(params=["inner"])
+def join_type(request):
+    return request.param
+
+
+@pytest.fixture(params=["zarr", "h5ad"])
+def file_format(request):
+    return request.param
+
+
+def _adatas_to_paths(adatas, tmp_path, file_format):
+    """
+    Gets list of adatas, writes them and returns their paths as zarr
+    """
+    paths = None
+
+    def write_func(adata, path):
+        if file_format == "h5ad":
+            adata.write(path)
+        else:
+            adata.write_zarr(path)
```

Comment:
But wouldn't that require checking for file format, opening a group according to the format, and then using write elem? Asking because I also would like to avoid that

Reply:
Why?

Though this isn't terribly important as it's just part of the test.

Reply:
I was worried that is was going to look cumbersome but I found a way to deal with it
```python
write_elem(as_group(p, mode='a'), '', a)
```
Which is good I suppose to test `as_group` too